lib/Quantization/Serialization.cpp

```diff
@@ -66,6 +70,13 @@ template <> struct ScalarTraits<FloatWrapper> {
   static QuotingType mustQuote(StringRef) { return QuotingType::None; }
 };
 
+/// Mapping for YAMLGlowToolsVersion yaml serializer.
+template <> struct MappingTraits<YAMLGlowToolsVersion> {
+  static void mapping(IO &io, YAMLGlowToolsVersion &ver) {
+    io.mapRequired("GlowToolsVersion", ver.value);
```

Comment:
@mciprian13  Actually -- does this mean that even if `GLOW_BUILD_DATE ` is not defined that this will be required to exist in the yaml, even though it'll be an empty string? Maybe we should make this optional?

Reply:
@jfix71 The ```GLOW_BUILD_DATE``` macro always exists in the upstream version. See the top glow `CMakeList.txt`:
```
# Compile definition for Glow build date in Y-M-D format.
string(TIMESTAMP GLOW_BUILD_DATE "%Y-%m-%d")
add_definitions("-DGLOW_BUILD_DATE=\"${GLOW_BUILD_DATE}\"")
```
I guess your internal build does not have the macro defined. But either way I think it is better to have the same YAML format for both upstream/internal builds in order to allow exchanging profiling infos between the tools. So I would suggest always dumping/reading the version whether it is an explicit date (upstream version) or empty string (for your internal build). Or maybe you could also define the macro on your side as well (if my understanding is correct).
WDYT?

Reply:
Ok, I don't mind it much being required in the yaml format, it's just that this is a breaking change (every time we update the yaml format, I mean). And since this was more for debugging purposes it doesn't seem strictly "required" -- but I don't have strong feelings here. (It's tangential as to whether we have it defined internally, if not defined it's just an empty string, but would still be required when loading the yaml file. I.e. I was more wondering on the required/optional side that makes it a breaking change for loading yaml files without the build date)